### DISCUSSION TITLE
V7 with shouldUnregister Controller after unmount/mount does not use defaultValues unlike in V6

### DISCUSSION THREAD

**Discussion by alexanderleonenko** [react-hook-form/react-hook-form#5651] [Q&A]
**Describe the bug**
V7 with `shouldUnregister: true` `Controller` after unmount/mount does not use `defaultValues` unlike in V6

**To Reproduce**

1. Go to v6 sandbox https://codesandbox.io/s/smoosh-hooks-t0n0j
2. Click on 'toggle' x2 times to hide/show `Controller`
4. Input's value is present
5. Go to v7 sandbox https://codesandbox.io/s/friendly-snow-312xq
6. Click on 'toggle' x2 times to hide/show `Controller`
8. Input's value is empty

**Codesandbox link (Required)**
v6: https://codesandbox.io/s/smoosh-hooks-t0n0j
v7: https://codesandbox.io/s/friendly-snow-312xq

V6:
![v6](https://user-images.githubusercontent.com/85992052/122551296-10491080-d035-11eb-8fa6-f2da309dc64e.gif)

V7:
![v7](https://user-images.githubusercontent.com/85992052/122551319-19d27880-d035-11eb-9707-f7d7bfa82af5.gif)

**Comment by JamesWong1999** [comment]
perhaps you want to use `shouldUnregister: false`? v7 will remove that input completely after inputs get unmounted, i think unless you leave a inline `defaultValue`.

**Reply by alexanderleonenko** [reply]
I have a lot of show/hide controls and I don't want to keep all values or isValid=true formState for unmounted Controllers.
`shouldUnregister: false` is not the best solution for me.
I just want to mount/unmount `Controller` multiple times and each time take value for it from `defaultValues`.
Btw it was fine in the version `7.8.3`.

@bluebill1049 could please suggest, should I use only Controller's prop `defaultValue` for that case and skip useForm's `defaultValue` since `7.8.4`?
